File: list_missing_cards.py
# python - filename: list_missing_cards.py
import json
from pathlib import Path

p = Path(r"c:\Users\sbawaney\Documents\VSCode Files\DeckOfAdventuring")
a = json.load(open(p / "5etools_DoMMT.json", "r", encoding="utf-8"))
b = json.load(open(p / "CardsJsons" / "AllCards.json", "r", encoding="utf-8"))

names_a = {c.get("name") for c in a.get("cards", []) if c.get("name")}
names_b = set(b.get("cards", {}).keys())

missing = sorted(names_a - names_b)
print("\n".join(missing) if missing else "None - all cards present in AllCards.json")

File: list_missing_cards.js
// node - filename: list_missing_cards.js
const path = require('path');
const a = require(path.resolve("c:/Users/sbawaney/Documents/VSCode Files/DeckOfAdventuring/5etools_DoMMT.json"));
const b = require(path.resolve("c:/Users/sbawaney/Documents/VSCode Files/DeckOfAdventuring/CardsJsons/AllCards.json"));

const namesA = new Set(a.cards.map(c => c.name).filter(Boolean));
const namesB = new Set(Object.keys(b.cards || {}));

const missing = [...namesA].filter(n => !namesB.has(n)).sort();
console.log(missing.length ? missing.join("\n") : "None - all cards present in AllCards.json");